File: api/src/main/java/com/sourceplusplus/api/model/trace/TraceOrderType.java
package com.sourceplusplus.api.model.trace;

/**
 * Different trace order types core can be queried by.
 *
 * @author <a href="mailto:brandon@srcpl.us">Brandon Fergerson</a>
 * @version 0.3.0
 * @since 0.1.0
 */
public enum TraceOrderType {

    LATEST_TRACES,
    SLOWEST_TRACES,
    FAILED_TRACES,
    TOTAL_TRACES
}

File: protocol/src/commonMain/kotlin/com/sourceplusplus/protocol/artifact/trace/TraceOrderType.kt
package com.sourceplusplus.protocol.artifact.trace

import com.sourceplusplus.protocol.artifact.OrderType

/**
 * todo: description.
 *
 * @since 0.1.0
 * @author [Brandon Fergerson](mailto:bfergerson@apache.org)
 */
enum class TraceOrderType : OrderType {
    LATEST_TRACES,
    SLOWEST_TRACES,
    FAILED_TRACES;

    //todo: not need to replace _TRACES?
    val id = name.replace("_TRACES", "").toLowerCase()
    override val description = id.toLowerCase().capitalize()
    val fullDescription = name.toLowerCase().split("_").joinToString(" ") { it.capitalize() }
}
